Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -700,13 +704,26 @@ class Gx2Fitter {
 
       // calculate delta params [a] * delta = b
       deltaParams = BoundVector::Zero();
-      const ActsVector<reducedMatrixSize> deltaParamsReduced =
-          aMatrix.topLeftCorner<reducedMatrixSize, reducedMatrixSize>()
-              .colPivHouseholderQr()
-              .solve(bVector.topLeftCorner<reducedMatrixSize, 1>());
-
-      for (size_t idp = 0; idp < reducedMatrixSize; idp++) {
-        deltaParams(idp, 0) = deltaParamsReduced(idp, 0);
+      if (gx2fOptions.zeroField) {
+        const size_t reducedMatrixSize = 4;
+        const ActsVector<reducedMatrixSize> deltaParamsReduced =
+            aMatrix.topLeftCorner<reducedMatrixSize, reducedMatrixSize>()
+                .colPivHouseholderQr()
```

Comment:
I didn't notice t his before, but what's this? 😄

Reply:
I am not sure which part you are unsure about:
- `colPivHouseholderQr()`: It is the standard decomposition for solving with eigen. Maybe in our case, there could be a more efficient one, but it definitely requires investigation. https://eigen.tuxfamily.org/dox/group__TutorialLinearAlgebra.html
- reducedMatrixSize: We need to reduce the system in case of missing B-Field, since we don't have QoP information. Otherwise the matrix equation would fail to zero-entries.
- Equation: aMatrix * deltaParams = bVector